pkg/zfs/resize.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package zfs
+
+import (
+	"github.com/Sirupsen/logrus"
+	apis "github.com/openebs/zfs-localpv/pkg/apis/openebs.io/core/v1alpha1"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"os/exec"
+)
+
+// ResizeExt4 can be used to run a resize command on the ext4 filesystem
+// to expand the filesystem to the actual size of the device
+func ResizeExt4(devpath string) error {
+	cmd := exec.Command("resize2fs", devpath)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		logrus.Errorf("zfspv: resize failed error: %s", string(out))
+		return err
+	}
+	return nil
+}
+
+// ResizeXFS can be used to run a resize command on the xfs filesystem
+// to expand the filesystem to the actual size of the device
+func ResizeXFS(path string) error {
+	cmd := exec.Command("xfs_growfs", path)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		logrus.Errorf("zfspv: resize failed error: %s", string(out))
+		return err
+	}
+	return nil
+}
+
+// HandleVolResize resizes the filesystem
```

Comment:
how does resize get handled for ZFS datasets?

Reply:
for zfs dataset, we just have to set the quota, nothing else needs to be done.